### PR TITLE
Emit metrics to cloudwatch on test execution status

### PR DESCRIPTION
**Pull request by vikasmb**
**What type of PR is this?**
feature

**Which issue does this PR fix**:
Emits metrics to cloudwatch regarding test status.

**What does this PR do / Why do we need it**:
Emits metrics to cloudwatch regarding test status.

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

Tested it locally.

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

No
**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:

No
**Does this PR introduce any user-facing change?**:

No
```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/lib/integration.sh [hunk 1]
```diff
@@ -23,6 +23,7 @@ function run_kops_conformance() {
     KOPS_TEST_DURATION=$((SECONDS - START))
     echo "TIMELINE: KOPS tests took $KOPS_TEST_DURATION seconds."
 
+    emit_cloudwatch_metric "kops_test_status" "1"
```
Comment (jayanthvn) [inline scripts/lib/integration.sh:26]: Can we also add for bottlerocket, calico and perf tests?

Comment (vikasmb) [reply scripts/lib/integration.sh]: Yes, updated it in next commit.

File: scripts/run-integration-tests.sh [hunk 2]
```diff
@@ -35,6 +35,7 @@ __cluster_deprovisioned=0
 
 on_error() {
     echo "Error with exit code $1 occurred on line $2"
+    emit_cloudwatch_metric "error_occurred" "1"
```
Comment (jayanthvn) [inline scripts/run-integration-tests.sh:38]: Here we won't know which test failed right? or do we get that granularity in CW?

Comment (jayanthvn) [reply scripts/run-integration-tests.sh:38]: Synced up with Vikas on this.

**Review by jayanthvn — APPROVED** [review]
lgtm

**Pull request merged**